src/main/java/com/archisacademy/jobportal/mapper/impl/CertificateMapperImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.archisacademy.jobportal.mapper.impl;
+
+import com.archisacademy.jobportal.dto.CertificateDto;
+import com.archisacademy.jobportal.loggers.MainLogger;
+import com.archisacademy.jobportal.mapper.CertificateMapper;
+import com.archisacademy.jobportal.model.Certificate;
+import com.archisacademy.jobportal.repositories.ProfileRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import java.sql.Timestamp;
+
+
+@Component
+public class CertificateMapperImpl implements CertificateMapper {
+    private final ProfileRepository profileRepository;
+    private final static MainLogger LOGGER = new MainLogger(CertificateMapperImpl.class);
+
+    public CertificateMapperImpl(ProfileRepository profileRepository) {
+        this.profileRepository = profileRepository;
+    }
+
+    @Override
+    public CertificateDto toDto(Certificate certificate) {
+        return CertificateDto.builder()
+                .name(certificate.getCertificateName())
+                .companyName(certificate.getCompanyName())
+                .postingDate(certificate.getPostingDate())
+                .hours(certificate.getCertificateHours())
+                .url(certificate.getCertificateUrl())
+                .profileId(certificate.getProfile().getId())
+                .build();
+    }
+
+    @Override
+    public Certificate toEntity(CertificateDto certificateDto) {
+        return Certificate.builder()
+                .certificateName(certificateDto.getName())
+                .companyName(certificateDto.getCompanyName())
+                .postingDate(new Timestamp(System.currentTimeMillis()))
+                .certificateHours(certificateDto.getHours())
+                .certificateUrl(certificateDto.getUrl())
+                .profile(profileRepository.findById(certificateDto.getProfileId())
```

Comment:
you should be fetching details inside mapper methods. please do map only the values which is coming from parameter object.